File: q1-q99/q50-q59/q53/q53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_max = nums[0]
        solution = nums[0]
        for i in range(1, len(nums)):
            if nums[i] > current_max + nums[i]:
                current_max = nums[i]
            else:
                current_max = current_max + nums[i]
            if current_max > solution:
                solution = current_max
        return solution

File: q1-q99/q50-q59/q53/q53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int current_max = nums[0];
        int solution = nums[0];
        for (int i = 1; i < nums.size(); i++){
            current_max = nums[i] > current_max + nums[i] ? nums[i] : current_max + nums[i];
            if (current_max > solution)
                solution = current_max;
        }
        return solution;
    }
};
